src/AppBundle/Controller/Report/ReportController.php

```diff
@@ -160,6 +160,42 @@ public function update(Request $request, $id)
             ]);
         }
 
+        if (array_key_exists('prof_deputy_other_costs', $data)) {
+            $defaultCostTypeIds = array_column($report->getProfDeputyOtherCostTypeIds(), 'typeId');
+
+            foreach ($data['prof_deputy_other_costs'] as $postedProfDeputyOtherCostType) {
+                if (in_array(
+                    $postedProfDeputyOtherCostType['prof_deputy_other_cost_type_id'],
+                        $defaultCostTypeIds
+                    )) {
+                    $profDeputyOtherCost = $report->getProfDeputyOtherCostByTypeId(
+                        $postedProfDeputyOtherCostType['prof_deputy_other_cost_type_id']
+                    );
+
+                    // update if exists, or instantiate a new entitys
+                    if ($profDeputyOtherCost instanceof EntityDir\Report\ProfDeputyOtherCost) {
+                        $profDeputyOtherCost->setAmount($postedProfDeputyOtherCostType['amount']);
+                    } else {
+                        $profDeputyOtherCost = new EntityDir\Report\ProfDeputyOtherCost(
+                            $report,
+                            $postedProfDeputyOtherCostType['prof_deputy_other_cost_type_id'],
+                            $postedProfDeputyOtherCostType['has_more_details'],
+                            $postedProfDeputyOtherCostType['amount']
+                        );
+                    }
+                    if ($profDeputyOtherCost->getHasMoreDetails()) {
+                        $profDeputyOtherCost->setMoreDetails($postedProfDeputyOtherCostType['more_details']);
+                    }
+
+                    $this->getEntityManager()->persist($profDeputyOtherCost);
+                }
+            }
+            $report->updateSectionsStatusCache([
+                Report::SECTION_PROF_DEPUTY_COSTS
+            ]);
+            $this->getEntityManager()->flush();
```

Comment:
Same here and a few other places : I would expect data to be flushed and then update cache but may not make a difference